src/cmd/flux-start.c

```diff
@@ -336,7 +336,7 @@ static void state_cb (flux_subprocess_t *p, flux_subprocess_state_t state)
                 log_msg ("%d (pid %d) %s", cli->rank, pid, strsignal (SIGCONT));
             else if (WIFSIGNALED (status))
                 log_msg ("%d (pid %d) %s", cli->rank, pid, strsignal (WTERMSIG (status)));
-            else if (WIFEXITED (status))
+            else if (WIFEXITED (status) && WEXITSTATUS (status) != 0)
```

Comment:
See the update in #1749. This function also needs the first erroneous `WIFSIGNALED` conditional removed, and there is no point in testing for `WIFCONTINUED` and `WIFSTOPPED`

Reply:
Good catch!  Fixed and squashed.